py-polars/polars/internals/construction.py

```diff
@@ -901,18 +828,239 @@ def sequence_to_pydf(
+        ]
+        return PyDataFrame(data_series)
+
+    raise ValueError(
+        f"orient must be one of {{'col', 'row', None}}, got {orient} instead."
+    )
+
+
+@_sequence_to_pydf_dispatcher.register(tuple)
+def _sequence_of_tuple_to_pydf(
+    first_element: tuple[Any, ...],
+    data: Sequence[Any],
+    schema: SchemaDefinition | None,
+    schema_overrides: SchemaDict | None,
+    orient: Orientation | None,
+    infer_schema_length: int | None,
+) -> PyDataFrame:
+    if is_namedtuple(first_element):
         if schema is None:
-            column_names = []
+            schema = first_element._fields  # type: ignore[attr-defined]
+            if len(first_element.__annotations__) == len(schema):
+                schema = [
+                    (name, py_type_to_dtype(tp, raise_unmatched=False))
+                    for name, tp in first_element.__annotations__.items()
+                ]
+        elif orient is None:
+            orient = "row"
+
+    return _sequence_of_sequence_to_pydf(
+        first_element,
+        data=data,
+        schema=schema,
+        schema_overrides=schema_overrides,
+        orient=orient,
+        infer_schema_length=infer_schema_length,
+    )
+
+
+def _sequence_of_series_to_pydf(
+    first_element: pli.Series,
+    data: Sequence[Any],
+    schema: SchemaDefinition | None,
+    schema_overrides: SchemaDict | None,
+    orient: Orientation | None,
+    infer_schema_length: int | None,
+) -> PyDataFrame:
+    series_names = [s.name for s in data]
+    column_names, schema_overrides = _unpack_schema(
+        schema or series_names,
+        schema_overrides=schema_overrides,
+        n_expected=len(data),
+    )
+    data_series: list[PySeries] = []
+    for i, s in enumerate(data):
+        if not s.name:  # TODO: Replace by `if s.name is None` once allowed
+            s.rename(column_names[i], in_place=True)
+        new_dtype = schema_overrides.get(column_names[i])
+        if new_dtype and new_dtype != s.dtype:
+            s = s.cast(new_dtype)
+        data_series.append(s._s)
+
+    data_series = _handle_columns_arg(data_series, columns=column_names)
+    return PyDataFrame(data_series)
+
+
+@_sequence_to_pydf_dispatcher.register(str)
+def _sequence_of_single_to_pydf(
+    first_element: Any,
+    data: Sequence[Any],
+    schema: SchemaDefinition | None,
+    schema_overrides: SchemaDict | None,
+    orient: Orientation | None,
+    infer_schema_length: int | None,
+) -> PyDataFrame:
+    column_names, schema_overrides = _unpack_schema(
+        schema, schema_overrides=schema_overrides, n_expected=1
+    )
+    data_series: list[PySeries] = [
+        pli.Series(column_names[0], data, schema_overrides.get(column_names[0]))._s
+    ]
+    data_series = _handle_columns_arg(data_series, columns=column_names)
+    return PyDataFrame(data_series)
+
+
+if _NUMPY_AVAILABLE:
+
+    @_sequence_to_pydf_dispatcher.register(np.ndarray)
+    def _sequence_of_ndarray_to_pydf(
```

Comment:
Gating functions behind an if-statement like this looks really weird to me. But then again, I don't know what a better solution would be.

Reply:
I tend to think of this as akin to a C-style `#ifdef`  :)

Reply:
> Gating functions behind an if-statement like this looks really weird to me. But then again, I don't know what a better solution would be.

I believe this could be rewritten:

```
def _sequence_of_ndarray_to_pydf(...): ... # same as in PR, but without the decorator

if _NUMPY_AVAILABLE:
    _sequence_to_pydf_dispatcher.register(np.ndarray, _sequence_of_ndarray_to_pydf)
```

I don't think the function will be useful unless numpy is available, though.